src/lib.rs

```diff
@@ -706,6 +726,94 @@ where
         }
     }
 
+    /// Removes multiple of a value from the bag. If `quantity` is greater than the number of
+    /// occurences, zero occurances will remain.
+    ///
+    /// The number of occurrences of the value currently in the bag is returned.
+    ///
+    /// The value may be any borrowed form of the bag's value type, but
+    /// [`Hash`] and [`Eq`] on the borrowed form *must* match those for
+    /// the value type.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use hashbag::HashBag;
+    ///
+    /// let mut bag = HashBag::new();
+    ///
+    /// bag.insert_many('x', 10);
+    /// assert_eq!(bag.contains(&'x'), 10);
+    /// assert_eq!(bag.remove_many(&'x', 3), 7);
+    /// assert_eq!(bag.contains(&'x'), 7);
+    /// assert_eq!(bag.remove_many(&'x', 10), 0);
+    /// ```
+    #[inline]
+    pub fn remove_many<Q: ?Sized>(&mut self, value: &Q, quantity: usize) -> usize
+    where
+        T: Borrow<Q>,
+        Q: Hash + Eq,
+    {
+        match self.items.get_mut(value) {
+            None => 0,
+            Some(n) if *n <= quantity => {
+                self.count -= 1;
+                self.items.remove(value);
+                0
+            }
+            Some(n) => {
+                *n -= quantity;
```

Comment:
I believe you also need to decrement `self.count` here (and we should add a test for that).

Reply:
We do? I thought `count` was used to answer the question "how many entries in the bag are non-zero". If so, then we don't need to decrement `count` here. This final branch is for the case when then you remove some number of items from the bag, but there will still be at least one of those items left.

Reply:
Ah, after a closer look, I see that I'm wrong. `count` is the sum of values in the bag, not the number of unique types of items. I'll correct this.